src/mesh/stl_io.C

```diff
@@ -0,0 +1,358 @@
+#else
+
+  // This is a serial-only process for now;
+  // the Mesh should be read on processor 0 and
+  // broadcast later
+  libmesh_assert_equal_to (MeshOutput<MeshBase>::mesh().processor_id(), 0);
+
+  // Clear the mesh so we are sure to start from a pristine state.
+  MeshBase & mesh = MeshInput<MeshBase>::mesh();
+  mesh.clear();
+  mesh.set_mesh_dimension(2);
+
+  std::string_view basename = Utility::basename_of(filename);
+  const bool gzipped_file = (basename.rfind(".gz") == basename.size() - 3);
+
+  std::unique_ptr<std::istream> file;
+
+  if (gzipped_file)
+    {
+#ifdef LIBMESH_HAVE_GZSTREAM
+      auto inf = std::make_unique<igzstream>();
+      libmesh_assert(inf);
+      inf->open(filename.c_str(), std::ios::in);
+      file = std::move(inf);
+#else
+      libmesh_error_msg("ERROR: need gzstream to handle .gz files!!!");
+#endif
+    }
+  else
+    {
+      auto inf = std::make_unique<std::ifstream>();
+      libmesh_assert(inf);
+
+      std::string new_name = Utility::unzip_file(filename);
+
+      inf->open(new_name.c_str(), std::ios::in);
+      file = std::move(inf);
+    }
+
+  const std::regex all_expected_chars
+    ("^[\\w\\d\\-\\.\\^\\$]*$");
+
+  const std::regex start_regex
+    ("^\\s*solid");
+  const std::regex start_with_name_regex
+    ("^\\s*solid\\s+(\\w+)");
+
+  const std::regex start_facet_regex
+    ("^\\s*facet");
+
+  // We'll ignore facets with normals for now
+  /*
+  const std::regex facet_with_normal_regex
+    ("^\\s*facet\\s+normal"
+     "\\s+([+-]?(\\d+([.]\\d*)?([eE][+-]?\\d+)?|[.]\\d+([eE][+-]?\\d+)?)"
+     "\\s+[+-]?(\\d+([.]\\d*)?([eE][+-]?\\d+)?|[.]\\d+([eE][+-]?\\d+)?)"
+     "\\s+[+-]?(\\d+([.]\\d*)?([eE][+-]?\\d+)?|[.]\\d+([eE][+-]?\\d+)?))");
+  */
+
+  const std::regex start_loop_regex
+    ("^\\s*outer\\s+loop");
+
+  const std::regex vertex_regex
+    ("^\\s*vertex"
+     "\\s+([+-]?(\\d+([.]\\d*)?([eE][+-]?\\d+)?|[.]\\d+([eE][+-]?\\d+)?)"
+     "\\s+[+-]?(\\d+([.]\\d*)?([eE][+-]?\\d+)?|[.]\\d+([eE][+-]?\\d+)?)"
+     "\\s+[+-]?(\\d+([.]\\d*)?([eE][+-]?\\d+)?|[.]\\d+([eE][+-]?\\d+)?))");
+
+  const std::regex end_facet_regex
+    ("^\\s*end\\s*facet");
+
+  const std::regex end_loop_regex
+    ("^\\s*end\\s*loop");
+
+  bool have_started = false;
+  bool in_facet = false;
+  bool in_vertex_loop = false;
+  std::unique_ptr<Tri3> triangle;
+  int next_vertex = 0;
+  int line_num = 0;
+
+  std::unordered_map<Point, Node *> mesh_points;
+
+  for (std::string line; std::getline(*file, line);)
+    {
+      ++line_num;
+      std::smatch sm;
+
+      for (char & c : line)
+        {
+          libmesh_error_msg_if
+            (c != '\n' && c != '\r' &&
+             (c < ' ' || c > '~'),
+             "Found non-ASCII character " << int(c) << " on line " <<
+             line_num << "\nSTLIO does not yet support binary files.");
+        }
+
+      if (std::regex_search(line, sm, start_regex))
+        {
+          if (std::regex_search(line, sm, start_with_name_regex))
```

Comment:
Our current `std::regex_search` configure test does not test the version of that [function](https://en.cppreference.com/w/cpp/regex/regex_search) taking a `std::match_results<T>`, so if we do decide to make `std::regex` required, we should probably also beef up the test a little, so it at least tests all the functionality we use.

Reply:
Oh, yikes, definitely.  I'm already using `match_results` in our new `SparseMatrix::read_matlab` too.